gpseer/maximum_likelihood.py

```diff
@@ -256,33 +384,56 @@ def main(
     model.add_gpm(gpm)
     logger.info("└──> Done constructing model.")
 
+    # Fit the model to the data
     logger.info("Fitting data...")
-    model.fit()
+
+    try:
+        model.fit()
+    except epistasis.models.utils.FittingError as err:
+
+        # Brittle check to see if this is the univariate spline error.  If it
+        # is, return a useful help message.
+        if err.args[0].startswith("scipy.interpolate.UnivariateSpline fitting returned more parameters"):
+            informative_err = "\n\nspline fit failed.  Try increasing --spline_smoothness\n"
+            raise RuntimeError(informative_err)
+
+        # If not, re-raise the error.  If our brittle check for the specific
+        # error fails, we'll still get an informative error back from epistasis
+        # (and sklearn/scipy)
+        raise err
+
     logger.info("└──> Done fitting data.")
 
+    # Figure out which genotypes to predict.
     genotypes_to_predict = None
     if genotype_file:
         genotypes_to_predict = read_genotype_file(wildtype, genotype_file)
 
+    # Do the actual prediction
     logger.info("Predicting missing data...")
     out_df = predict_to_dataframe(
         model,
         genotypes_to_predict=genotypes_to_predict,
     )
     logger.info("└──> Done predicting.")
 
-    # Figure out the root for any output graphs
-    out_root = output_file.split(".")
-    if out_root[-1] in ["csv","txt","text","xls","xlsx"]:
-        out_root = ".".join(out_root[:-1])
-    else:
-        out_root = output_file
-
-    # Plot pdfs of diagnostic graphs
-    plots_to_pdf(model,out_df,out_root)
+    # Calculate some stats
+    logger.info("Calculating fit statistics...")
+    stats_df, convergence_df = create_stats_output(model)
+    stats_df.to_csv("{}_fit-information.csv".format(output_root))
+    convergence_df.to_csv("{}_convergence.csv".format(output_root))
+    logger.info("└──> Done.")
```

Comment:
What about printing this info through the logger?

Reply:
you mean print the convergence stats directly?

Reply:
Yeah—that way, they can see convergence stats in their console. I'd rather not open the file if I don't have to.

Reply:
Good idea.  Will implement on this PR.

Reply:
Okay, added better logging.  Here's example output: 

![image](https://user-images.githubusercontent.com/4952186/79276642-3f765780-7e5d-11ea-8e84-8f1c5dbc8428.png)

Also made a few more tweaks:
+ cross-validate and estimate-ml now use the same function (in utils) to check for existing files.
+ cross-validate now has `cross_validate_to_dataframe` call that is equivalent tot the ml `predict_to_dataframe`.